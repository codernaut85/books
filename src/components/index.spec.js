import TestRenderer from 'react-test-renderer';
import PaginationComponent from './Pagination';


describe("PaginationComponent", () => {  
  it("renders correctly when there are very few books", () => {
    const testRenderer = TestRenderer.create(<PaginationComponent bookCount={30} page={1} />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it("renders correctly when there are many books", () => {
    const testRenderer = TestRenderer.create(<PaginationComponent bookCount={2500} page={10} />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});