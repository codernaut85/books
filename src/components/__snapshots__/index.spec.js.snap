// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationComponent renders correctly when there are many books 1`] = `
<div
  className="pagination-controls"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/1"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
        <span
          className="sr-only"
        >
          First
        </span>
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/9"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span
          aria-hidden="true"
        >
          ‹
        </span>
        <span
          className="sr-only"
        >
          Previous
        </span>
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/1"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/2"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/3"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        3
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/4"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        4
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/5"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        5
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/6"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        6
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/7"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        7
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/8"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        8
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/9"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        9
      </a>
    </li>
    <li
      className="page-item active"
    >
      <span
        className="page-link"
        disabled={false}
        href="/books/page/10"
      >
        10
        <span
          className="sr-only"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/11"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        11
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/12"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        12
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/13"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        13
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/14"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        14
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/15"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        15
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/16"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        16
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/17"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        17
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/18"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        18
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/19"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        19
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/20"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        20
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/11"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span
          aria-hidden="true"
        >
          ›
        </span>
        <span
          className="sr-only"
        >
          Next
        </span>
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/125"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
        <span
          className="sr-only"
        >
          Last
        </span>
      </a>
    </li>
  </ul>
</div>
`;

exports[`PaginationComponent renders correctly when there are very few books 1`] = `
<div
  className="pagination-controls"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/1"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
        <span
          className="sr-only"
        >
          First
        </span>
      </a>
    </li>
    <li
      className="page-item active"
    >
      <span
        className="page-link"
        disabled={false}
        href="/books/page/1"
      >
        1
        <span
          className="sr-only"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/2"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/2"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span
          aria-hidden="true"
        >
          ›
        </span>
        <span
          className="sr-only"
        >
          Next
        </span>
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="/books/page/2"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
        <span
          className="sr-only"
        >
          Last
        </span>
      </a>
    </li>
  </ul>
</div>
`;
